src/Microsoft.Identity.Web/TokenCacheProviders/MsalAbstractTokenCacheProvider.cs

```diff
@@ -66,15 +95,36 @@ private async Task OnAfterAccessAsync(TokenCacheNotificationArgs args)
             }
         }
 
+        private byte[] ProtectBytes(byte[] msalBytes)
+        {
+            if (msalBytes != null && _protector != null)
+            {
+                return _protector.Protect(msalBytes);
+            }
+
+            return msalBytes!;
+        }
+
         private static CacheSerializerHints CreateHintsFromArgs(TokenCacheNotificationArgs args) => new CacheSerializerHints { CancellationToken = args.CancellationToken, SuggestedCacheExpiry = args.SuggestedCacheExpiry };
 
         private async Task OnBeforeAccessAsync(TokenCacheNotificationArgs args)
         {
             if (!string.IsNullOrEmpty(args.SuggestedCacheKey))
             {
                 byte[] tokenCacheBytes = await ReadCacheBytesAsync(args.SuggestedCacheKey, CreateHintsFromArgs(args)).ConfigureAwait(false);
-                args.TokenCache.DeserializeMsalV3(tokenCacheBytes, shouldClearExistingCache: true);
+
+                args.TokenCache.DeserializeMsalV3(UnprotectBytes(tokenCacheBytes), shouldClearExistingCache: true);
```

Comment:
What's the intended behavior/exception during failure to Unprotect the bytes? Should it be the same as during failure to retrieve the cache key, or should there be a more strict failure?

Reply:
fixed. great catch. thank you